src/redux/preview.js

```diff
@@ -763,10 +763,11 @@ module.exports.getRemixes = id => (dispatch => {
     });
 });
 
-module.exports.getProjectStudios = id => (dispatch => {
+module.exports.getProjectStudios = (id, ownerUsername, isAdmin, token) => (dispatch => {
     dispatch(module.exports.setFetchStatus('projectStudios', module.exports.Status.FETCHING));
     api({
-        uri: `/projects/${id}/studios`
+        uri: `${isAdmin ? '/admin' : `/users/${ownerUsername}`}/projects/${id}/studios`,
```

Comment:
Hmmm wouldn't this cause the admin route to just be `/admin` instead of what I assume should be `/admin/users/:name/projects/:id/studios`?

Reply:
Oh wait misread the string interpolation. I thought we were doing admin routes just by prefixing the normal route with `admin`, but it looks like here admin route doesn't have the same route (doesn't have /users/:user). Is that right? Is that supposed to be that way?

Reply:
Yes, admin and non-admin routes are supposed to be different in this way.

The idea with the separate admin routes is that we can skip a bunch of checking about the user's permission to view this particular resource; if the user is indeed an admin, we just let them view it.

We're adding the resource's owner's username to non-admin routes so that we can do a standardized first pass permissions check; the basic logic is "is the user the owner? if so, then proceed. if not, is the resource publicly visible/shared? if so, then proceed."

Reply:
Ah ok i see.